ui/app/components/license-banners.js

```diff
@@ -9,11 +17,25 @@
  * @param {string} expiry - RFC3339 date timestamp
  */
 
-import Component from '@glimmer/component';
-import isAfter from 'date-fns/isAfter';
-import differenceInDays from 'date-fns/differenceInDays';
-
 export default class LicenseBanners extends Component {
+  @service version;
+
+  @tracked currentVersion = this.version.version;
+  @tracked warningDismissed;
+  @tracked expiredDismissed;
+
+  constructor() {
+    super(...arguments);
+    // If nothing is saved in localStorage or the user has updated their Vault version, do not dismiss any of the banners.
+    const localStorageLicenseBannerObject = localStorage.getItem('licenseBanner');
+    if (!localStorageLicenseBannerObject || localStorageLicenseBannerObject.version !== this.currentVersion) {
+      localStorage.setItem('licenseBanner', { dismissType: '', version: this.currentVersion });
```

Comment:
I think the logic here could be simplified by storing the dismissal at a variable key, rather than storing an object that has to be parsed and reasoned about. So for example, if I'm on version 1.12.2, this component would check for a value (any value, really) at `licenseBanner-warning-1.12.2`
The get and set calls would be able to use a key variable that is calculated based on the version:
```
const warningBannerKey = `licenseBanner-warning-${this.currentVersion}`
const dismissedWarning = localStorage.getItem(warningBannerKey)
```

Reply:
I like the idea. Let me know what you think of the changes. Not exactly your suggestion, but a similar concept.